torch_frame/nn/utils/loss.py

```diff
@@ -0,0 +1,11 @@
+import torch
+from torch import Tensor
+
+
+def cross_entropy_with_logits(
+    logits: Tensor,
+    targets: Tensor,
+) -> Tensor:
+    r"""Cross Entropy Loss for Soft Multi-classification Labels."""
+    probs = torch.softmax(logits, dim=1)
+    return -(targets * probs.log()).sum(1).mean()
```

Comment:
Thank you for the instant feedback!

Yes, I see, the `F.cross_entropy` and `CrossEntropyLoss` have supported the soft targets since PyTorch `1.10.0`, while I kept the habit of manually realizing it as in the old versions, I have removed the file `torch_frame/nn/utils/loss.py`, it is indeed unnecessary, thank you!

For four mixup types (`none`, `feature`, `hidden`, `ordinary`), the `ordinary` is the previous mixup version implemented in this project, thus I left that option in the codes. In the paper we propose `feature` & `hidden` and both are always better than `ordinary`. `none` represents no mixup, and sometimes there are few datasets preferring this option, it depends on inherent data characteristics. **As you suggested, I have removed the `ordinary` option**. Since other options are totally compatible with the advanced features of `torch.compile`, **I uniformly integrate them into `forward` function and remove `foward_mixup` which was for the `ordinary` option, if the user needs mixup during training, just simply calling `model(tf, mixup_encoded=True)`, and during inference without mixup, call `model(tf)`.**

The related documentation is detailed and fixed, and `torch.compile` testings are passed.

One more thing, I found my pull requests cannot pass the pytest under PyTorch `1.12.0`, `1.13.0` and `2.0.0` with a consistent single failed case here, while I configured the same environments as action files (I'm not sure the local ones are exactly the same as the used ones in GitHub) with all these versions no problem. The failed case here is `batch_size = 0`, `mixup = 'hidden'`, `task_type = 'binary_classification'`, it seems everything in real usage (`batch_size != 0`) or in my local machine (`batch_size == 0`) is all right.

Reply:
Thank you for addressing the comments!

Reply:
> One more thing, I found my pull requests cannot pass the pytest under PyTorch 1.12.0, 1.13.0 and 2.0.0 with a consistent single failed case here, while I configured the same environments as action files (I'm not sure the local ones are exactly the same as the used ones in GitHub) with all these versions no problem. The failed case here is batch_size = 0, mixup = 'hidden', task_type = 'binary_classification', it seems everything in real usage (batch_size != 0) or in my local machine (batch_size == 0) is all right.

Apologies. That's a flaky test on our side. Fixed in https://github.com/pyg-team/pytorch-frame/pull/392